data/users/externalldap.py

```diff
@@ -147,11 +147,21 @@ def _get_ldap_referral_dn(self, referral_exception):
         referral_dn = referral_uri[len("ldap:///") :]
         return referral_dn
 
+    def _clean_user_filter(self, query):
+        if not query:
+            return None
+
+        if self._ldap_user_filter[0] == "(":
```

Comment:
Check for the ")" on the other end too and if not present, fail with `None`

Reply:
This will be picked up by the verification on the ldap side. This is mostly just to catch the cases where user puts in something just as a straight string like "someOtherField=someOtherValue". Anything more complicated than that we'll leave to the validation.